### PR TITLE
IA-651 autocomplete

### PR DESCRIPTION
**Pull request by beygorghor**
- removing react-select deps,
- rewrite select custom with autocomplete from material  ui

https://user-images.githubusercontent.com/12494624/125947385-e7c26ad2-7807-48d8-b05d-45d01ea1c591.mov

### CODE REVIEW

**Comment by quang-le** [comment]
Had to run `npm ci --legacy-peer-deps` to be able to update the dependencies. The problem was coming from `react-table`,  which uses a version that doesn't support react 17

File: src/components/inputs/Select/index.js [hunk 1]
```diff
@@ -1,123 +1,208 @@
...
+                });
+                setMultiSelectedValue(newSelectedValue);
+            } else {
+                const newSelectedValue = getOption(value);
+                if (newSelectedValue) setSelectedValue(newSelectedValue);
+            }
+        } else {
+            multi ? setMultiSelectedValue([]) : setSelectedValue(null);
+        }
+    }, [value, options]);
```
Comment (quang-le) [inline src/components/inputs/Select/index.js:67]: `multi` should be included in the dependencies array

File: src/components/inputs/Select/index.js [hunk 2]
```diff
@@ -1,123 +1,208 @@
...
+        if (value) {
+            if (multi) {
+                const newSelectedValue = [];
+                const valuesList = Array.isArray(value)
+                    ? value
+                    : value.split(',');
 
+                valuesList.forEach(v => {
+                    const option = getOption(v);
+                    if (option) newSelectedValue.push(getOption(v));
```
Comment (quang-le) [inline src/components/inputs/Select/index.js:57]: Why not ` if (option) newSelectedValue.push(option);`?

File: src/components/inputs/Select/index.js [hunk 3]
```diff
@@ -1,123 +1,208 @@
...
+    }, [value, options]);
+
+    const handleChange = (e, newValue) => {
+        if ((!multi && !newValue) || (multi && newValue.length === 0)) {
+            return onChange(null);
+        }
+        if (multi) {
+            return onChange(newValue.map(v => v && v.value).join(','));
+        }
+        return onChange(newValue.value);
```
Comment (quang-le) [inline src/components/inputs/Select/index.js:76]: Theoretically, I thiink this should be wrapped in a `useCallback` with `multi` in the dependencies array, although I agree it's unlikely `multi` will change

File: src/components/inputs/Select/index.js [hunk 4]
```diff
@@ -1,123 +1,208 @@
...
+                        if (!option) return null;
+                        return (
+                            <Chip
+                                classes={{
+                                    label: classes.chipLabel,
+                                }}
+                                color="secondary"
+                                label={option.label}
+                                {...getTagProps({ index })}
+                            />
```
Comment (quang-le) [inline src/components/inputs/Select/index.js:109]: The callback is big enough to be extracted I think

File: src/components/inputs/Select/index.js [hunk 5]
```diff
@@ -1,123 +1,208 @@
...
+                            style: { color: 'transparent' },
+                        };
+                        inputExtraProps.startAdornment = (
+                            <div className={classes.startAdornment}>
+                                {extraProps.renderOption({
+                                    label: params.inputProps.value,
+                                })}
+                            </div>
+                        );
+                        paramsCopy.inputProps.value = '';
```
Comment (quang-le) [inline src/components/inputs/Select/index.js:136]: Big enough callback to be extracted

**Review by quang-le — CHANGES_REQUESTED** [review]
Most comments are minor changes, but in `/Select/index.js`, line 67, `multi` should be added to the deps array

**Review by quang-le — APPROVED** [review]
I approve in order not to block merging while I'm away, but please make the change requested in my previous review

File: src/components/inputs/Select/index.js [hunk 6]
```diff
@@ -1,123 +1,208 @@
...
+    const [selectedValue, setSelectedValue] = useState(null);
+    const [multiSelectedValue, setMultiSelectedValue] = useState([]);
     const intl = useSafeIntl();
-    const [focus, setFocus] = useState(isFocused);
+    const classes = useStyles();
+
+    const getOption = optionValue =>
+        options.find(o => `${o.value}` === `${optionValue}`);
+
+    useEffect(() => {
```
Comment (olethanh) [inline src/components/inputs/Select/index.js:47]: I made the change requested by Son, but I'm actually not sure why this part is in a useEffect and not part of the handleChange? wouldn't that trigger a double render?

File: src/components/inputs/Select/index.js [hunk 7]
```diff
@@ -1,123 +1,207 @@
...
+                startAdornment: (
+                    <div className={classes.startAdornment}>
+                        {extraProps.renderOption({
+                            label: params.inputProps.value,
+                        })}
+                    </div>
+                ),
+                style: { color: 'transparent' },
+            };
+            inputExtraProps.startAdornment = (
```
Comment (olethanh) [inline src/components/inputs/Select/index.js:104]: probably the wrong key name, as it the same as same as the one just above?

File: src/components/inputs/Select/index.js (hunk 6 above)
Comment (beygorghor) [reply src/components/inputs/Select/index.js:47]: This is a controlled input, the handleChange is only there to adapt the behaviour on change, useEffect is getting the correct value.

File: src/components/inputs/Select/index.js (hunk 7 above)
Comment (beygorghor) [reply src/components/inputs/Select/index.js:104]: the second one is useless

**Pull request merged**